### PR TITLE
Fixed races in ratelimiter tests on go1.5

### LINKED ISSUE #7785: pkg/util/ratelimiter	race with go1.5
When running test I'm getting following race warning:
```
==================
WARNING: DATA RACE
Read by goroutine 16:
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction.func3()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:51 +0x87

Previous write by goroutine 14:
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:46 +0x3df
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:456 +0xdc

Goroutine 16 (running) created at:
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:53 +0x3b5
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:456 +0xdc

Goroutine 14 (running) created at:
  testing.RunTests()
      /usr/lib/golang/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/lib/golang/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/openshift/origin/pkg/util/ratelimiter/_test/_testmain.go:100 +0x384
==================
==================
WARNING: DATA RACE
Write by goroutine 14:
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:35 +0x1af
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:456 +0xdc

Previous read by goroutine 31:
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction.func3()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:48 +0x37

Goroutine 14 (running) created at:
  testing.RunTests()
      /usr/lib/golang/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/lib/golang/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/openshift/origin/pkg/util/ratelimiter/_test/_testmain.go:100 +0x384

Goroutine 31 (finished) created at:
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:53 +0x3b5
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:456 +0xdc
==================
==================
WARNING: DATA RACE
Write by goroutine 21:
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction.func2()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:38 +0x54
  github.com/openshift/origin/pkg/util/ratelimiter.(*RateLimitedFunction).handleOne()
      github.com/openshift/origin/pkg/util/ratelimiter/_test/_obj_test/ratelimiter.go:56 +0x9a
  github.com/openshift/origin/pkg/util/ratelimiter.(*RateLimitedFunction).RunUntil.func1()
      github.com/openshift/origin/pkg/util/ratelimiter/_test/_obj_test/ratelimiter.go:48 +0xaf
  k8s.io/kubernetes/pkg/util/wait.JitterUntil.func1()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:66 +0x58
  k8s.io/kubernetes/pkg/util/wait.JitterUntil()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:67 +0x7d
  k8s.io/kubernetes/pkg/util/wait.Until()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/wait/wait.go:47 +0x4b

Previous read by goroutine 14:
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:58 +0x4b2
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:456 +0xdc

Goroutine 21 (running) created at:
  github.com/openshift/origin/pkg/util/ratelimiter.(*RateLimitedFunction).RunUntil()
      github.com/openshift/origin/pkg/util/ratelimiter/_test/_obj_test/ratelimiter.go:48 +0xd2
  github.com/openshift/origin/pkg/util/ratelimiter.TestRateLimitedFunction()
      /home/maszulik/workspace/origin/src/github.com/openshift/origin/pkg/util/ratelimiter/ratelimiter_test.go:44 +0x319
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:456 +0xdc

Goroutine 14 (running) created at:
  testing.RunTests()
      /usr/lib/golang/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/lib/golang/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/openshift/origin/pkg/util/ratelimiter/_test/_testmain.go:100 +0x384
==================
PASS
coverage: 91.7% of statements
Found 3 data race(s)
FAIL	github.com/openshift/origin/pkg/util/ratelimiter	15.027s
```

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #7785. This bites me heavily, so I've decided to fix it.
@ramr ptal

### CODE REVIEW

**Comment by ramr** [comment]
LGTM

File: pkg/util/ratelimiter/ratelimiter_test.go [hunk 1]
```diff
@@ -33,28 +52,25 @@ func TestRateLimitedFunction(t *testing.T) {
...
-				if tc.Interval > 0 {
+			go func(rlf *RateLimitedFunction, idx, interval int) {
+				if interval > 0 {
 					rlf.Invoke(rlf)
 				} else {
-					rlf.Invoke(i)
+					rlf.Invoke(idx)
 				}
-			}()
+			}(rlf, i, tc.Interval)
```
Comment (ramr) [inline pkg/util/ratelimiter/ratelimiter_test.go]: Don't really need this as what we are only interested in is that Invoke is called - the parameter is not really used specifically for this test. But its ok to put this change in.

File: pkg/util/ratelimiter/ratelimiter_test.go [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+	h._counter += 1
+	return nil
+}
+
+func (h *handler) counter() int {
+	h.Lock()
+	defer h.Unlock()
+	return h._counter
+}
+
```
Comment (ramr) [inline pkg/util/ratelimiter/ratelimiter_test.go]: thanks for taking care of this and the lock. +1

**Comment by ramr** [comment]
LGTM

**Comment by soltysh** [comment]
[merge]

**Pull request merged**